File: Retos/Reto #4 - PRIMO, FIBONACCI Y PAR [Media]/python/EspinoLeandroo.py
def primo(n):
    if n <= 1:
        return False
    
    for i in range(2,n):
        if n % i == 0:
            return False

    return True


def es_fibonacci(n):
    num1 = 0
    num2 = 1
    counter = 0

    while num1 <= n:
        if num1 == n:
            return True

        num3 = num2 + num1
        num1 = num2
        num2 = num3
        counter += 1
    
    return False


def es_par(n):
    return n % 2 == 0

n =  int(input("Ingresa Número: "))

message = "El número " + str(n)

if primo(n):
    message += " es primo, "
else:
    message += " no es primo, "

if es_fibonacci(n):
    message += "es fibonacci, "
else:
    message += "no es fibonacci, "

if es_par(n):
    message += "es par."
else:
    message += "es impar."


print(message)

File: Retos/Reto #4 - PRIMO, FIBONACCI Y PAR [Media]/java/EspinoLeandroo.java
import java.util.Scanner;

public class EspinoLeandroo {

    public static void main(String[] args) {
        EspinoLeandroo espinoLeandroo = new EspinoLeandroo();

        Scanner sc = new Scanner(System.in);

        System.out.print("Ingresa Número: ");
        int n = sc.nextInt();

        System.out.print("El número " + n);
        if (espinoLeandroo.primo(n)) {
            System.out.print(" es primo, ");
        } else {
            System.out.print(" no es primo, ");
        }
        if (espinoLeandroo.esFibonacci(n)) {
            System.out.print("es fibonacci, ");
        } else {
            System.out.print("no es fibonacci, ");
        }
        if (espinoLeandroo.esPar(n)) {
            System.out.print("es par.");
        } else {
            System.out.print("es impar.");
        }

    }

    private boolean esPar(int n) {
        return n % 2 == 0;
    }

    private boolean esFibonacci(int n) {
        int num1 = 0, num2 = 1;
        int counter = 0;

        while (num1 <= n) {
            if (num1 == n) {
                return true;
            }

            int num3 = num2 + num1;
            num1 = num2;
            num2 = num3;
            counter = counter + 1;
        }
        return false;
    }

    private boolean primo(int n) {
        if (n <= 1)
            return false;

        for (int i = 2; i < n; i++)
            if (n % i == 0)
                return false;

        return true;
    }

}
